x-pack/plugins/stack_connectors/server/connector_types/bedrock/bedrock.ts

```diff
@@ -125,33 +151,95 @@ export class BedrockConnector extends SubActionConnector<Config, Secrets> {
+    const signed = this.signRequest(
+      formattedBody,
+      `/model/${model}/invoke-with-response-stream`,
+      true
+    );
+
+    const response = await this.request({
+      ...signed,
+      url: `${this.url}/model/${model}/invoke-with-response-stream`,
+      method: 'post',
+      responseSchema: StreamingResponseSchema,
+      data: formattedBody,
+      responseType: 'stream',
+    });
+
+    return response.data.pipe(new PassThrough());
+  }
+
+  public async invokeStream(body: InvokeAIActionParams): Promise<Transform> {
+    const res = (await this.streamApi({
+      body: JSON.stringify(body),
+      stream: true,
+    })) as unknown as IncomingMessage;
+    return res.pipe(new PassThrough()).pipe(transformToString());
+  }
+
+  /**
+   * Deprecated. Use invokeStream instead.
+   * TODO: remove before 8.12 FF
+   * No token tracking implemented for this method
    */
   public async invokeAI({
     messages,
     model,
   }: InvokeAIActionParams): Promise<InvokeAIActionResponse> {
-    const combinedMessages = messages.reduce((acc: string, message) => {
-      const { role, content } = message;
-      // Bedrock only has Assistant and Human, so 'system' and 'user' will be converted to Human
-      const bedrockRole = role === 'assistant' ? '\n\nAssistant:' : '\n\nHuman:';
-      return `${acc}${bedrockRole}${content}`;
-    }, '');
-
-    const req = {
-      // end prompt in "Assistant:" to avoid the model starting its message with "Assistant:"
-      prompt: `${combinedMessages} \n\nAssistant:`,
-      max_tokens_to_sample: DEFAULT_TOKEN_LIMIT,
-      temperature: 0.5,
-      // prevent model from talking to itself
-      stop_sequences: ['\n\nHuman:'],
-    };
-
-    const res = await this.runApi({ body: JSON.stringify(req), model });
+    const res = await this.runApi({ body: JSON.stringify(formatBedrockBody({ messages })), model });
     return { message: res.completion.trim() };
   }
 }
+
+const formatBedrockBody = ({
+  messages,
+}: {
+  messages: Array<{ role: string; content: string }>;
+}) => {
+  const combinedMessages = messages.reduce((acc: string, message) => {
+    const { role, content } = message;
+    // Bedrock only has Assistant and Human, so 'system' and 'user' will be converted to Human
+    const bedrockRole = role === 'assistant' ? '\n\nAssistant:' : '\n\nHuman:';
+    return `${acc}${bedrockRole}${content}`;
+  }, '');
+
+  return {
+    // end prompt in "Assistant:" to avoid the model starting its message with "Assistant:"
+    prompt: `${combinedMessages} \n\nAssistant:`,
+    max_tokens_to_sample: DEFAULT_TOKEN_LIMIT,
+    temperature: 0.5,
+    // prevent model from talking to itself
+    stop_sequences: ['\n\nHuman:'],
+  };
+};
+
+const transformToString = () =>
+  new Transform({
+    transform(chunk, encoding, callback) {
+      const encoder = new TextEncoder();
+      const decoder = new EventStreamCodec(toUtf8, fromUtf8);
```

Comment:
I could not find documentation on how to handle their codec. I did find https://github.com/awsdocs/aws-doc-sdk-examples/issues/5541 for when documentation should be added.

I followed the bedrock extension in the langchainjs repo here: https://github.com/langchain-ai/langchainjs/pull/2367/files#diff-ead6be41360b3bf1c4559ea4d0069a015b69a57d5f8c51d0cce73253fde30aa2R246-R268